ddl/partition.go

```diff
@@ -1813,6 +1822,22 @@ func (w *worker) onDropTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (
 				return ver, errors.Trace(err)
 			}
 		}
+		if tblInfo.TiFlashReplica != nil {
```

Comment:
Could you explain a bit more? I didn't get the idea here...

Reply:
I could not find that dropped partitions were ever removed from the AvailablePartitionIDs, so I added it here, since when dropping (or cleaning up after a failed add or reorganize partition) those ids should no longer be in that list.
Not sure if this explains it?
Or how is the AvailablePartitionIDs list handled in this case?

Reply:
I think I got your point, and I think it makes sense.
However, I am wondering why the system can work before this and what will be changed after this. I have not modified these codes, and I found they exist in release-5.4 which is a very old version, so they actually work for a long time.
IMO, they can work because this is a drop action, a partition is dropped, and will not be accessed then. However, its data will not be deleted until being deleted by the gc worker. During the time gap, if we must access the partition, they are actually available in TiFlash. And if we do a flashback, they will be available again(maybe some tests could be added here). So the author have written like this.
I think it could be better if we can ask the author for a comfirmation.

Reply:
> I think it could be better if we can ask the author for a comfirmation.

@crazycs520 Could you please take a look?

Reply:
I think it's reasonable to delete dropped partition id from `TiFlashReplica.AvailablePartitionIDs` here. You can find similar logic in `onTruncateTablePartition` function too.

I didn't delete the dropped partition id in `tblInfo.TiFlashReplica.AvailablePartitionIDs` before in `onDropTablePartition ` function because it didn't affect the correctness.  I think this was my previous oversight, even if it didn't affect correctness. 

BTW, I think the name of the variable `physicalTableIDs` here is misleading, here `physicalTableIDs` actually only contains the partition id which in dropping, not all partition's ID.